### PR TITLE
Modify Section Hierarchy in Automate Code Deploys with CI/CD

### LINKED ISSUE #1882: Heading hierarchy on CI/CD page is wrong
**Affected Document:**

https://github.com/astronomer/docs/blob/main/astro/ci-cd.md

**Problem Type (Priority):**

- [ ] Error (Critical): This document has existing information that's incorrect and prevents readers from completing the documented setup (Add the `error/critical` label to this issue)
- [x] Misleading (Important): This document could benefit from additional context, guidance, or clarity (Add the `error/FYI` label to this issue)
- [ ] Typo (Low): Any issue related to spelling/ punctuation/ formatting (Add the `error/typo` label to this issue)

**Suggested Change (include links to relevant existing docs, support tickets, messages, or external content):**

The hierarchy and therefore the indented structure of the TOC is misleading. It looks like all the CI/CD options (Jenkins, CodeBuild etc) are children of "Deploy with Github Actions":

image

**Additional Notes:**

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #1882

### CODE REVIEW

**Comment by stuart23** [comment]
This looks better thanks @kwcanuck for jumping on it so quickly. Is there a separator you could put in the TOC so the provider examples are separated from the first two sections? If not, I think this is fine to merge.

**Review by stuart23 — APPROVED** [review]

File: astro/ci-cd.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 sidebar_label: 'CI/CD Templates'
 title: 'Automate code deploys with CI/CD'
 id: ci-cd
+toc_min_heading_level: 2
+toc_max_heading_level: 2
```
Comment (jwitz) [inline astro/ci-cd.md:6]:
```suggestion
toc_max_heading_level: 3
```
I think we should show the subtopics for GitHub in the sidebar

**Review by jwitz — CHANGES_REQUESTED** [review]
Thanks for taking this on @kwcanuck . The new header sizes look good, but can we go back to the original organization, where GitHub stays at the top of the list? I understand why you'd want to alphabetize, but GitHub is far and away the most common tool people will use to set up CI/CD, and I think it's best for us to keep that up front instead of burying it in the middle of the doc. It looks like the Azure CI/CD docs have a similar structure:
Screenshot 2023-02-13 at 1 39 29 PM

Let me know if there's any other edits besides the alphabetization here that I should take a closer look at (it's hard to tell from the "Files changed" tab as it stands)

File: astro/ci-cd.md (hunk 1 above)
Comment (kwcanuck) [reply astro/ci-cd.md:6]: @jwitz if you implement this change, a standalone Prerequisites heading appear in the right TOC. It looks much better as is.

Comment (jwitz) [reply astro/ci-cd.md:6]: Should the standalone prerequisites actually be there though, or should it be a level 4 header?

**Comment by kwcanuck** [comment]
@jwitz can you unblock this? I think it's worth merging as is. With the right TOC, users can find what they need quickly. The GitHub content is as easy to locate as it was previously, and the alphabetical organization will be quickly understood by most users.

If you disagree, feel free to implement the changes you feel are necessary. It would be great if we could get this merged. 

Thanks!

**Review by jwitz — APPROVED** [review]
✅ I reorganized the the doc by level of support/ templating we offer for each tool. Especially with the introduction of the new deploy GH Action, we can implicitly recommend that users try implementing CI/CD with with more supported tools by placing them higher in the TOC.

**Pull request merged**